pylint/checkers/base/basic_checker.py

```diff
@@ -453,7 +463,20 @@ def visit_expr(self, node: nodes.Expr) -> None:
 
         # Warn W0133 for exceptions that are used as statements
         if isinstance(expr, nodes.Call):
-            inferred = utils.safe_infer(expr)
+            inferred = None
+            if self._heuristic_exception_detection is True:
+                name, _, _ = expr.as_string().partition("(")
+                exc_suffixes = ("Error", "Exc", "Exception", "Warning")
+                if any(name.endswith(suffix) for suffix in exc_suffixes):
+                    inferred = utils.safe_infer(expr)
+                else:
+                    skipped = expr.as_string()
+                    warnings.warn(
+                        f"Skipped W0133 check for {skipped!r} because heuristic exception detection is enabled",
+                        RuntimeWarning,
+                    )
```

Comment:
That will be emitted for every instance and every run. It'll probably just spam the user. I'd recommend we remove it.